autometa/taxonomy/ncbi.py

```diff
@@ -188,96 +184,82 @@ def get_lineage_dataframe(self, taxids, fillna=True):
             how='left',
             left_on=<taxid_column>,
             right_index=True)
-
         """
         canonical_ranks = [r for r in reversed(NCBI.CANONICAL_RANKS)]
         canonical_ranks.remove('root')
         taxids = list(set(taxids))
-        taxids_ = {}
+        ranked_taxids = {}
         for rank in canonical_ranks:
             for taxid in taxids:
                 name = self.name(taxid, rank=rank)
-                if taxid not in taxids_:
-                    taxids_.update({taxid: {rank: name}})
+                if taxid not in ranked_taxids:
+                    ranked_taxids.update({taxid: {rank: name}})
                 else:
-                    taxids_[taxid].update({rank: name})
-        df = pd.DataFrame(taxids_).transpose()
+                    ranked_taxids[taxid].update({rank: name})
+        df = pd.DataFrame(ranked_taxids).transpose()
         df.index.name = 'taxid'
         if fillna:
             df.fillna(value='unclassified', inplace=True)
         return df
 
     def rank(self, taxid):
-        """Short summary.
+        """
+        Takes a taxid and returns its rank
 
         Parameters
         ----------
         taxid : int
-            `taxid` to retrieve rank from `nodes`.
+            identifer for a taxon in the Taxonomy Database by NCBI
+            `taxid` to retrieve rank from `nodes` or `merged`
 
         Returns
         -------
         str
-            rank name
-
-        Raises
-        -------
-        ValueError
-            Provided `taxid` is not an integer
-
+            rank name if taxid is found in nodes.dmp or merged.dmp else 'unclassified'
         """
-        # If taxid is not found in nodes.dmp, will return 'unclassified'
-        try:
-            taxid = int(taxid)
-        except ValueError as err:
-            logger.error(
-                f'Taxid must be an integer! {taxid} type --> {type(taxid)}')
-            return None
-        return self.nodes.get(taxid, {'rank': 'unclassified'}).get('rank')
+        self.is_valid_taxid(taxid)
+        rank_name = self.nodes.get(taxid, {'rank': 'unclassified'}).get('rank')
+        if rank_name == {'rank': 'unclassified'}:
+            return self.merged.get(taxid, {'rank': 'unclassified'}).get('rank')
+        return rank_name
```

Comment:
Since you are going through both `self.nodes` and `self.merged`, I'm not sure you should be setting a default value at the first lookup. The default is only necessary once. So if neither `self.nodes` or `self.merged` return a value, then you could return `'unclassified'`. I'm showing a ternary operator below

```python
rank_name = self.merged.get(taxid, {'rank': None}).get('rank')
return rank_name if rank_name else 'unclassified'
```